### PR TITLE
Copy the regexes to the public interface

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

**Comment by MorrisJobke** [comment]
👍

**Comment by icewind1991** [comment]
Wouldn't it be better to have an `isUserAgent` method (which accepts `IRequest::ANDROID`, etc) instead of exposing the implentation details of detection

**Comment by nickvergessen** [comment]
We have a method `isUserAgent` which accepts an array of regexes. The problem is, you don't have the regexes publicly available to use them on this method.

I would also only consider the name public api, not the actual value?

**Comment by MorrisJobke** [comment]
> I would also only consider the name public api, not the actual value?

Could this be done by an abstract constant? Is this possible?

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -67,10 +67,20 @@ class Request implements \ArrayAccess, \Countable, IRequest {
 	// Android Chrome user agent: https://developers.google.com/chrome/mobile/docs/user-agent
 	const USER_AGENT_ANDROID_MOBILE_CHROME = '#Android.*Chrome/[.0-9]*#';
 	const USER_AGENT_FREEBOX = '#^Mozilla/5\.0$#';
+	const REGEX_LOCALHOST = '/^(127\.0\.0\.1|localhost)$/';
```
Comment (ChristophWurst) [inline lib/private/AppFramework/Http/Request.php]: I know this is unrelated to your PR, but what about IPv6?

**Comment by ChristophWurst** [comment]
> Could this be done by an abstract constant? Is this possible?

No, according to http://php.net/manual/en/language.oop5.interfaces.php:
> It's possible for interfaces to have constants. Interface constants works exactly like class constants except they cannot be overridden by a class/interface that inherits them.

File: lib/private/AppFramework/Http/Request.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/AppFramework/Http/Request.php]: Open an ticket ;)

**Comment by nickvergessen** [comment]
I mean we could also add methods that then use `isUserAgent(self::CONST)` but I prefer those 3 consts over 3 new methods.

**Comment by LukasReschke** [comment]
👍

**Pull request merged**